### ISSUE TITLE
[adguard-home] Chart install fails with ingresses enabled

### ISSUE THREAD

**Issue by bpetrikovics** [rm3l/helm-charts#33]
First of all, thanks for putting together this chart!

When installing the chart with any of the ingresses enabled, it fails with the following error:

```
Error: INSTALLATION FAILED: unable to build kubernetes objects from release manifest: error parsing : invalid Yaml document separator: apiVersion: networking.k8s.io/v1
```

Inspecting the issue with `helm template` shows it's due to whitespace issue:

```
# Source: adguard-home/templates/ingresses.yaml
---apiVersion: networking.k8s.io/v1
kind: Ingress
```

A newline is missing at the beginning of the YAML document. It's apparently related to this block:

```
---
{{- if .Values.ingresses.https.enabled -}}
{{- $fullName := include "adguard-home.fullname" . -}}
{{- $svcPort := .Values.services.https.port -}}
{{- if and .Values.ingresses.https.className (not (semverCompare ">=1.18-0" .Capabilities.KubeVersion.  {{- if not (hasKey .Values.ingresses.https.annotations "kubernetes.io/ingress.class") }}
  {{- $_ := set .Values.ingresses.https.annotations "kubernetes.io/ingress.class" .Values.ingresses.ht  {{- end }}
{{- end }}
{{- if semverCompare ">=1.19-0" .Capabilities.KubeVersion.GitVersion -}}
apiVersion: networking.k8s.io/v1
{{- else if semverCompare ">=1.14-0" .Capabilities.KubeVersion.GitVersion -}}
apiVersion: networking.k8s.io/v1beta1
{{- else -}}
apiVersion: extensions/v1beta1
{{- end }}
kind: Ingress
```

The cause seems like that these lines consume newlines both at the beginning and the end of the lines, so for example the following would work:

```
---
{{- if .Values.ingresses.https.enabled -}}
{{- $fullName := include "adguard-home.fullname" . -}}
{{- $svcPort := .Values.services.https.port -}}
{{- if and .Values.ingresses.https.className (not (semverCompare ">=1.18-0" .Capabilities.KubeVersion.  {{- if not (hasKey .Values.ingresses.https.annotations "kubernetes.io/ingress.class") }}
  {{- $_ := set .Values.ingresses.https.annotations "kubernetes.io/ingress.class" .Values.ingresses.ht  {{- end }}
{{- end }}
{{ if semverCompare ">=1.19-0" .Capabilities.KubeVersion.GitVersion -}}
apiVersion: networking.k8s.io/v1
{{ else if semverCompare ">=1.14-0" .Capabilities.KubeVersion.GitVersion -}}
apiVersion: networking.k8s.io/v1beta1
{{ else -}}
apiVersion: extensions/v1beta1
{{- end }}
kind: Ingress
```

Thanks!

**Comment by rm3l** [comment]
Thanks for reporting this issue, @bpetrikovics ! Working on a fix.

**Comment by rm3l** [comment]
FYI, this issue should be fixed now. A new version of this Chart has been published accordingly: [0.8.1](https://github.com/rm3l/helm-charts/releases/tag/adguard-home-0.8.1).

Please do let me know if this issue still persists on your side.
